core/metrics/meter.go

```diff
@@ -0,0 +1,70 @@
+package metrics
+
+import (
+	"fmt"
+	"go.opentelemetry.io/otel"
+	stdout "go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
+	"go.opentelemetry.io/otel/metric"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.12.0"
+	"time"
+)
+
+// InitMeter creates and sets a global meter provider.
+func InitMeter(serviceName string, interval time.Duration) error {
+	// TODO configure exporter how we need.
+
+	exporter, err := stdout.New()
+	if err != nil {
+		return fmt.Errorf("creating exporter failed %w", err)
+	}
+	resource := resource.NewWithAttributes(
+		semconv.SchemaURL,
+		semconv.ServiceNameKey.String(serviceName),
+	)
+
+	mp := sdkmetric.NewMeterProvider(
+		sdkmetric.WithResource(resource),
+		sdkmetric.WithReader(
+			sdkmetric.NewPeriodicReader(exporter, sdkmetric.WithInterval(interval)),
+		),
+	)
+	otel.SetMeterProvider(mp)
+	return nil
+}
+
+// NewCounter creates a new meter counter instrument.
+// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#counter
+func NewCounter(meterName string, counterName string, desc string, units string) (metric.Int64Counter, error) {
```

Comment:
tbh, figure out which of these you need first and start integrating them into scribe or wherever you want and then we'll start figuring this out

Reply:
what do you think about
1. Use counters for reorg events (by chain)
2. Use a histogram for frequency of last block indexed (by chain, and use contract address as an attr). that way we can do max, min, and count on the blocks stored (max will tell us the max block height of that interval)

Reply:
like this to start actually. Let's get one of these meters up and then iterate from there